name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVImporter.java

```diff
@@ -290,6 +294,162 @@ public M parseObject(String source, ParsePosition pos)
         }
     }
 
+    public static class ISINField extends CSVImporter.Field
+    {
+
+        /* package */ ISINField(String name)
+        {
+            super(name);
+        }
+
+        public ISINFormat createFormat(List<Security> securityList)
+        {
+            return new ISINFormat(securityList);
+        }
+    }
+
+    public static class ISINFormat extends Format
+    {
+        private static final long serialVersionUID = 1L;
+        private List<String> isinList = new ArrayList<String>();
+        private static final String ISINpattern = "[A-Z]{2}[A-Z0-9]{9}\\d";
+
+        public ISINFormat(List<Security> securityList)
+        {
+            for (Security security : securityList)
+            {
+                if (security.getIsin() != null)
+                {
+                    String ISIN = security.getIsin().trim();
+                    if (CheckISIN(ISIN))
+                    {
+                        this.isinList.add(ISIN);
+                    }
+                }
+            }
+        }
+
+        @Override
+        public StringBuffer format(Object obj, StringBuffer toAppendTo, FieldPosition pos)
+        {
+            String s = (String)obj;
+            if (s == null)
+                throw new IllegalArgumentException();
+
+            return toAppendTo.append(s);
+        }
+
+        static boolean CheckISIN(String isin)
+        {
+            isin = isin.trim().toUpperCase();
+            if (!isin.matches("^"+ ISINpattern + "$"))
+                return false;
+            return Isin.isValid(isin);
+        }
+
+        @Override
+        public Object parseObject(String source, ParsePosition pos)
+        {
+            int parseSuccessIndex = source.length();
+            source = source.trim();
+            if (pos == null)
+                throw new NullPointerException();
+            Object rObj = null;
+
+            Pattern pattern = Pattern.compile(" (" + ISINpattern + ") ");
+            Matcher matcher = pattern.matcher(source);
+            boolean success = false;
+            if (matcher.find())
+            {
+                source  = matcher.group(1);
+            }
+
+            if (CheckISIN(source))
+            {
+                if (isinList.contains((String) source))
+                {
+                    success = true;
+                    rObj = (Object) source.toString();
+                }
+            }
+            if (success)
+                pos.setIndex(parseSuccessIndex);
+            return rObj;
+        }
+    }
+
+    public static final class HeaderSet
```

Comment:
Die Klassen HeaderSet und Header werden nirgendwo im Code verwendet. Ich vermute die kommen von einer zukünftigen Change. Lege die doch in eine Git Stash. Ungenutzter Code "auf Vorrat" macht nur die Maintenance schwerer.

Reply:
Abend Andreas,
Good catch, ja das Codesegment hatte ich vergessen für diesen Commit zu entfernen.
Habe eine weitere Änderung, mit der man beim Import Header, Typedefinition, Skiplines und Dateiformat vordefinieren kann...
Werde ich bei Gelegenheit mal als Pullrequest zusammenstellen.
Für den Moment habe ich diese Definition mit 855db4c entfernt.
Schönen Abend,
cmoaling

Reply:
> Habe eine weitere Änderung, mit der man beim Import Header, Typedefinition, Skiplines und Dateiformat vordefinieren kann...

Sowas hatte ich mir auch schon mal überlegt: das man die CSV Konfiguration auch als "Template" speichern kann. Und beim nächsten Import genau diese Konfiguration wieder auswählen kann (vielleicht vergibt man einen Namen und sieht es dann im Menü?). Allerdings würde ich vorschlagen wir ziehen diese Konfiguration dann nicht *in* die CSVImporter rein - sondern konfigurieren das von "außen". Separate Change.